crates/diman_lib/src/dimension_exponent.rs

```diff
@@ -2,6 +2,13 @@ use core::ops::{AddAssign, Mul, Neg};
 
 use crate::magnitude::Magnitude;
 
+// For some reason, this use statement is never recognized as used
+// even when I build the crates with no std, where removing the use
+// statement means that powi below cannot be used.
+#[allow(unused)]
+#[cfg(not(any(feature = "std")))]
+use num_traits::float::FloatCore;
```

Comment:
A minimized version of this issue: https://github.com/rust-lang/rust/issues/127852#issuecomment-2278496055